spring-cloud-alibaba-sentinel/src/main/java/com/alibaba/cloud/sentinel/feign/SentinelFeign.java

```diff
@@ -82,20 +83,24 @@ public InvocationHandler create(Target target,
 							"fallback");
 					Class fallbackFactory = (Class) getFieldValue(feignClientFactoryBean,
 							"fallbackFactory");
-					String name = (String) getFieldValue(feignClientFactoryBean, "name");
+					String beanName = (String) getFieldValue(feignClientFactoryBean,
+							"contextId");
+					if (!StringUtils.hasText(beanName)) {
```

Comment:
I don't think we need to check the beanName from `contextId`, see https://github.com/spring-cloud/spring-cloud-openfeign/blob/master/spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignClientsRegistrar.java#L283.

if contextId is empty, it will get by `getName` method.

Reply:
I see this code, in g version ,   it will get by getName method and will get serviceId. As I do this , it compatible f version and g version

Reply:
ok, well done, and we need to modify the code in finchley & 1.x branch 😿